sklearn/linear_model/_theil_sen.py

```diff
@@ -322,6 +322,18 @@ class TheilSenRegressor(RegressorMixin, LinearModel):
     array([-31.5871...])
     """
 
+    _parameter_constraints = {
+        "fit_intercept": ["boolean"],
+        "copy_X": ["boolean"],
+        "max_subpopulation": [Interval(Integral, 1, None, closed="left")],
+        "n_subsamples": [None, Integral],
+        "max_iter": [Interval(Integral, 0, None, closed="left")],
```

Comment:
Is there a compelling reason to allow for `max_iter=0`? That entirely skips the `_spatial_median` op and goes straight for the `ConvergenceWarning`.

```suggestion
        "max_iter": [Interval(Integral, 1, None, closed="left")],
```

Reply:
We're having this discussion with @glemaitre. We'd like to unify the meaning of max_iter=0. To me it should be valid and mean return the initialization.

Anyway, if it was valid before this PR, I think we should not break backward compatibility and still keep it valid in this PR.